databricks_cli/unity_catalog/api.py

```diff
@@ -221,9 +222,20 @@ def list_providers(self):
     def get_provider(self, name):
         return self.client.get_provider(name)
 
-    def update_provider(self, name, new_name=None, comment=None, recipient_profile=None):
-        return self.client.update_provider(name, new_name, comment, recipient_profile)
-
+    def update_provider(self, name, new_name=None, comment=None,
+                        recipient_profile=None, provider_spec=None):
+        # If spec is provided, override all legacy parameters.
+        if provider_spec is not None:
+            return self.client.update_provider(name, provider_spec)
+        _data = {}
+        if new_name is not None:
+            _data['name'] = new_name
+        if recipient_profile is not None:
+            _data['recipient_profile_str'] = mc_pretty_format(recipient_profile)
+        if comment is not None:
+            _data['comment'] = comment
```

Comment:
Are we keeping the `new_name`, `comment`, `recipient_profile` args for backward compatibility (in case other apps call this function)?  Looks like our CLI code only calls it with `provider_spec`.

Reply:
Yes. The API is "public", so the test has been complaining about backwards incompatibility. This is to maintain the existing API.